test/colony-network-auction.js

```diff
@@ -304,6 +307,450 @@ contract("Colony Network Auction", accounts => {
+        await forwardTime(duration, this);
+
+        const bidAmount = new BN(10).pow(new BN(30));
+        await giveUserCLNYTokens(colonyNetwork, BIDDER_1, bidAmount);
+        await clnyToken.approve(tokenAuction.address, bidAmount, { from: BIDDER_1 });
+        await tokenAuction.bid(bidAmount, { from: BIDDER_1 });
+
+        const bid = await tokenAuction.bids(BIDDER_1);
+        // Expect up to 1% error margin because of forwarding block time inaccuracies
+        const errorMarginRemainingToEndAuction = auctionProp.remainingToEndAuction.divn(100);
+        // Chai assert.closeTo does not work with Big Numbers so some manual comaring to error margin is required
+        const differenceRemainingToEndAuction = auctionProp.remainingToEndAuction.sub(bid);
+        // Check only the amount required to end the auction was accepted
+        expect(differenceRemainingToEndAuction).to.be.lte.BN(errorMarginRemainingToEndAuction);
+
+        await tokenAuction.finalize();
+        const finalPrice = await tokenAuction.finalPrice();
+        // Expect up to 1% error margin because of forwarding block time inaccuracies
+        const errorMarginFinalPrice = auctionProp.finalPrice.divn(100);
+        const differenceFinalPrice = auctionProp.finalPrice.sub(finalPrice);
+        // Check only the amount required to end the auction was accepted
+        expect(differenceFinalPrice).to.be.lte.BN(errorMarginFinalPrice);
+
+        await tokenAuction.claim(BIDDER_1);
+        const otherTokenBidderBalance = await otherToken.balanceOf(BIDDER_1);
+        expect(otherTokenBidderBalance, "the one bidder didn't receive all tokens").to.eq.BN(auctionProp.claimAmount);
+      });
+    });
+
+    it("auction closes when the receivedTotal goes over the total amount to end the auction", async () => {
+      const totalAmount = new BN(10).pow(new BN(16));
+      await otherToken.mint(totalAmount);
+      await otherToken.transfer(colonyNetwork.address, totalAmount);
+      const { logs } = await colonyNetwork.startTokenAuction(otherToken.address);
+      const auctionAddress = logs[0].args.auction;
+      tokenAuction = await DutchAuction.at(auctionAddress);
+
+      await giveUserCLNYTokens(colonyNetwork, BIDDER_1, new BN(10).pow(new BN(36)).muln(3));
+      await clnyToken.approve(tokenAuction.address, new BN(10).pow(new BN(36)).muln(3), { from: BIDDER_1 });
+
+      let endTime = await tokenAuction.endTime();
+      const amount = new BN(10).pow(new BN(20));
+
+      while (endTime.isZero()) {
+        await forwardTime(SECONDS_PER_DAY * 3, this);
+
+        await tokenAuction.bid(amount, { from: BIDDER_1 });
+        endTime = await tokenAuction.endTime();
+      }
+
+      await tokenAuction.finalize();
+      const finalPrice = await tokenAuction.finalPrice();
+      expect(finalPrice).to.eq.BN(new BN("40000000000000000000000"));
+    });
+
+    it("functions correctly even when price has reached the minimum", async () => {
```

Comment:
This isn't the scenario tested in this test, which has led to a race condition and the flickery test. If 10**16 tokens are minted, then the minimum price is 100. The price starts at 10**36, and goes down by an order of magnitude each day, so I'd expect the minimum price to only be reached at 34 days (and then only one person to be able to bid 1 CLNY wei before the auction closed).

Reply:
I've named the test more appropriately now and changed the calculations to be dynamic so to avoid marginal inaccuracies.